src/main/java/seedu/momentum/logic/commands/EditCommand.java

```diff
@@ -88,13 +89,20 @@ public CommandResult execute(Model model) throws CommandException {
      * Creates and returns a {@code Project} with the details of {@code projectToEdit}
      * edited with {@code editProjectDescriptor}.
      */
-    private static Project createEditedProject(Project projectToEdit, EditProjectDescriptor editProjectDescriptor) {
+    private static Project createEditedProject(Project projectToEdit,
+                                               EditProjectDescriptor editProjectDescriptor) throws ParseException {
         assert projectToEdit != null;
 
         Name updatedName = editProjectDescriptor.getName().orElse(projectToEdit.getName());
         Description updatedDescription = editProjectDescriptor.getDescription().orElse(projectToEdit.getDescription());
         Date createdDate = projectToEdit.getCreatedDate();
         Deadline updatedDeadline = editProjectDescriptor.getDeadline().orElse(projectToEdit.getDeadline());
+        if (editProjectDescriptor.getDeadline().isPresent()
+                && !Deadline.isBeforeCreatedDate(updatedDeadline.getDate().toString(), createdDate)) {
```

Comment:
I feel like ParseException should not be thrown here, perhaps a CommandException can be thrown so that we do not need to introduce ParseException to the execute method? What do the others think?

Reply:
I'm not sure, both will have the same behaviour but its more of a parse error than unable to execute command error since it is related to validity, but this will make the overall design better?

Reply:
Anyways I decided to change it

Reply:
Alright, noted! Thanks